### PR TITLE
V2 Settings - Security View

### LINKED ISSUE #3784: 2.0 Settings / Security {View}
[View in Figma](https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS-v2.0?node-id=1562%3A71532)

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

This PR adds two screens to the security section of the settings for V2.
- Change password
- Enable 2FA

Note that the settings shell is still a WIP so the pages will be missing headers.

Fixes #3784 

**Environment**: Staging(main branch) / Production

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- Visit http://localhost:3000/v2/settings/security/password
	 - Enter an invalid password, an error should appear
 	- Enter a valid password, it should be changed

- Visit http://localhost:3000/v2/settings/security/two-factor-auth
	- Enable 2FA
	- Disable 2FA

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/web/components/v2/settings/DisableTwoFactorModal.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import { SyntheticEvent, useState } from "react";
```
Comment (joeauyeung) [inline apps/web/components/v2/settings/DisableTwoFactorModal.tsx:1]: Copied this file but added V2 components

File: apps/web/components/v2/settings/EnableTwoFactorModal.tsx [hunk 2]
```diff
@@ -0,0 +1,232 @@
+import React, { SyntheticEvent, useState } from "react";
```
Comment (joeauyeung) [inline apps/web/components/v2/settings/EnableTwoFactorModal.tsx:1]: Also copied this file and added V2 components

File: packages/trpc/server/routers/viewer/auth.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    await prisma.user.update({
+      where: {
+        id: user.id,
+      },
+      data: {
+        password: hashedPassword,
+      },
+    });
+  },
+});
```
Comment (joeauyeung) [inline packages/trpc/server/routers/viewer/auth.tsx:63]: Moved changing a password from an API endpoint to tRPC.

File: packages/trpc/server/routers/viewer/auth.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      throw new TRPCError({ code: "BAD_REQUEST", message: "INCORRECT_PASSWORD" });
+    }
+
+    if (oldPassword === newPassword) {
+      throw new TRPCError({ code: "BAD_REQUEST", message: "PASSWORD_MATCHES_OLD" });
+    }
+
+    if (!validPassword(newPassword)) {
+      throw new TRPCError({ code: "BAD_REQUEST", message: "INVALID_PASSWORD" });
+    }
```
Comment (joeauyeung) [inline packages/trpc/server/routers/viewer/auth.tsx:51]: Checks for a valid password that was not present in the API endpoint

**Comment by zomars** [comment]
@joeauyeung don't forget to add these pages to the `middleware.ts`

**Review by zomars — COMMENTED** [review]
Looking good @joeauyeung let's address some possible changes incoming in #3817 first  🙏🏽

**Review by zomars — APPROVED** [review]

**Pull request merged**